modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/liferay-clay.tld

```diff
@@ -432,7 +438,7 @@
 			<rtexprvalue>true</rtexprvalue>
 		</attribute>
 		<attribute>
-			<description></description>
+			<description>Deprecated as of 7.4.0, with no direct replacement.</description>
```

Comment:
@jbalsas asked if we needed to keep it. 

So my 2 cents here would be that we have two options: 

**Keep it**

Keep it because accessibility and use it in the same way is being used in `RadioTag`.

This makes me thing in two suboptions:

- We would need then to add `showLabel` behavior to clay component, because right now if you render a `Radio` or a `Checbox` with no label they won't be accessible if you don't pass `aria-label`.
- We process it in the Tag so if `showLabel == true` we set `ariaLabel = label` and pass it as a prop to the react component, so we won't need to add nothing to clay.

**Deprecate it**

We can  use dynamicAttributes to get `ariaLabel` in the tag, so we leave the responsibility of making the component accesible to the user. Then we just need to pass that `ariaLabel` as a prop to the react component.

Reply:
I'd say remove it for now and then see usages and figure out what needs to be done.

Reply:
@kresimir-coko if we're deprecating `showLabel` we need to add the `setter` to the tag and mark it as deprecated. As long as an attribute is present in the tld a setter is needed.

Reply:
Okay, just the `setter` and nothing else? It doesn't need the getter? Is that because if someone passes it as a prop we want to consume it by setting it, and then just ignore it, hence the lack of getter (and code that uses it).

Reply:
Something (don't remember if the compiler, the source formatter or jvm) complains if no setter is defined for an existing attribute

Reply:
@carloslancha I'm looking for examples of deprecating the `setter` but I can't find an example that deprecates only the `setter`, all of the tags that deprecate a `setter` they also deprecate the `getter` too, and they add it to the list of private properties and they clean the up in the `cleanUp` method. For example the `monospaced`  property here in the [IconTag](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/IconTag.java#L76)

Reply:
@kresimir-coko  Take into account that we are creating a new java class `CheckboxTag`. So we don't need to deprecate any method that was present in the old `soy.CheckboxTag`. The only reason to add `showLabel` setter is because that attribute is present in the `tld` and we need to add it for the tag to not crash.  But because we're also deprecating that attribute in the tld we can deprecate the setter. We don't need to deprecate the getter because it wasn't there before (again, this is a new class). I know it's a little confusing... 😅 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2FYl8afftEGE03C%2Fgiphy.gif&token=groucho" data-gitmeme-token="groucho"><img src="https://media3.giphy.com/media/Yl8afftEGE03C/giphy.gif" title="Created by gitme.me with /groucho"/></a>

Reply:
Is `showLabel` used anywhere? I don't think the missing setter will fail unless used. So, if it's not used currently anywhere, just simply avoid adding it.

Reply:
I see, the reason why I even started checking other examples is because the module isn't building.
Here's how I did it:
```java
/**
 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
 */
@Deprecated
public void setShowLabel(boolean showLabel) {
	_showLabel = showLabel;
}
```

The error I'm getting:
```bash
CheckboxTag.java:98: error: cannot find symbol
                _showLabel = showLabel;
                ^
symbol:   variable _showLabel
location: class CheckboxTag
```
Was the body of the `setShowLabel` supposed to be different?

Reply:
> Is showLabel used anywhere? I don't think the missing setter will fail unless used in runtime. So, if it's not used currently anywhere, just simply avoid adding it.

It's acceptable that an attribute that's present in the tag definition, even if it's marked as deprecated, makes the tag fail if you pass it? If it's acceptable, yes, just get rid of the `setter`, if not, what I said in my previous comments.

Reply:
> The error I'm getting:

That error is because you didn't define `_showLabel` variable (in the end of the file, with the rest of `private Type Name` stuff

Reply:
> Is it acceptable that an attribute that's present in the tag definition, even if it's marked as deprecated, makes the tag fail if you pass it? If it's acceptable, yes, just get rid of the setter, if not, what I said in my previous comments.

That's why I mentioned usage. If you were using it before, then it's totally unacceptable. If you weren't and now for whatever reason you decide to start using an already deprecated method that does nothing... then I think that getting an error is not so bad...

Reply:
And the additional assumption we are making is that we're mostly the only ones consuming these tags.

Reply:
> That's why I mentioned usage. If you were using it before, then it's totally unacceptable. If you weren't and now for whatever reason you decide to start using an already deprecated method that does nothing... then I think that getting an error is not so bad...

I take this as *don't add `showLabel` to the tag* , because we aren't using the `<clay:checkbox>` anywhere in the portal (except the clay-example module).

Reply:
> I take this as don't add showLabel to the tag , because we aren't using the <clay:checkbox> anywhere in the portal (except the clay-example module).

Ok, if there're no usages let's keep `showLabel` in the `tld` marking it as deprecated on behalf of the good practices and just burn everything else 😅 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia4.giphy.com%2Fmedia%2FB0yHMGZZLbBxS%2Fgiphy.gif&token=burn" data-gitmeme-token="burn"><img src="https://media4.giphy.com/media/B0yHMGZZLbBxS/giphy.gif" title="Created by gitme.me with /burn"/></a>